src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/params.py

```diff
@@ -0,0 +1,47 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from ._util import ParametersContext
+
+with ParametersContext(command='monitor alert-rules') as c:
+    c.register_alias('name', ('--azure-resource-name',))
+    c.register_alias('rule_name', ('--name', '-n'))
```

Comment:
The `--name` parameters here need a values completer within their scope (except for create commands).